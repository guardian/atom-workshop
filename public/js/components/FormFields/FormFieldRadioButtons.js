import React, {PropTypes} from 'react';
import ShowErrors from '../Utilities/ShowErrors';
import { errorPropType } from '../../constants/errorPropType';

export default class FormFieldRadioButtons extends React.Component {

  static propTypes = {
    fieldLabel: PropTypes.string,
    fieldName: PropTypes.string,
    selectValues: PropTypes.array.isRequired,
    fieldValue: PropTypes.string,
    fieldErrors: PropTypes.arrayOf(errorPropType),
    onUpdateField: PropTypes.func
  };

  onUpdate = (e) => {
    this.props.onUpdateField(e.target.value);
  }

  renderButton(value, i) {
    return (
      <div key={i} className="form__group form__group--radio">
        <input className="form__radio-btn" type="radio" name={this.props.fieldName} value={value} checked={this.props.fieldValue === value ? 'checked' : false} onChange={this.onUpdate} />
        <span className="form__label form__label--radio">{value}</span>
      </div>
    );
  }

  renderButtons() {
    return this.props.selectValues.map(this.renderButton, this);
  }

  render() {
    return (
        <div>
          <label className="form__label" htmlFor={this.props.fieldName}>{this.props.fieldLabel}</label>
          {this.renderButtons()}
          {this.props.fieldErrors && this.props.fieldErrors.length ? <ShowErrors errors={this.props.fieldErrors}/>  : false}
        </div>
    );
  }
}
