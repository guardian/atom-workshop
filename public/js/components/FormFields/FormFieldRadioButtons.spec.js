import React from 'react';
import FormFieldRadioButtons from './FormFieldRadioButtons';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let fieldLabel = 'test',
    fieldName = 'test',
    fieldValues = ['One', 'Two'];

test('Should render', () => {

  const updateFn = jest.fn();
  const component = renderer.create(
    <FormFieldRadioButtons fieldLabel={fieldLabel} fieldName={fieldName} selectValues={fieldValues} onUpdateField={updateFn} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should call update function on change', () => {

  const updateFn = jest.fn();
  const radio = shallow(
    <FormFieldRadioButtons fieldLabel={fieldLabel} fieldName={fieldName} selectValues={fieldValues} onUpdateField={updateFn} />
  );

  radio.find('input').first().simulate('change', {target: {value: "test"}});

  expect(updateFn).toHaveBeenCalledTimes(1);
});
