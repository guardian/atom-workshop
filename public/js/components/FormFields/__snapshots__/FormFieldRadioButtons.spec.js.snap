exports[`test Should render 1`] = `
<div
  className="form__row">
  <label
    className="form__label"
    htmlFor="test">
    test
  </label>
  <div
    className="form__group form__group--radio">
    <input
      checked={false}
      className="form__radio-btn"
      name="test"
      onChange={[Function]}
      type="radio"
      value="One" />
    <span
      className="form__label form__label--radio">
      One
    </span>
  </div>
  <div
    className="form__group form__group--radio">
    <input
      checked={false}
      className="form__radio-btn"
      name="test"
      onChange={[Function]}
      type="radio"
      value="Two" />
    <span
      className="form__label form__label--radio">
      Two
    </span>
  </div>
</div>
`;
